### PR TITLE
V11 - Remove functions node bundle

### PR DESCRIPTION
**Pull request by DellaBitta**
### Discussion

With the removal of fetch we no long need to create a node bundle for functions. Instead the sourcebase may become isomorphic, so long as we remove the older node sources. And that's what the PR does!

### Testing

 - ts-node scripts/ci-test/check-paths.ts
 - CI.

### API Changes

N/A

### CODE REVIEW

File: packages/functions-compat/package.json
Comment (hsubox76) [inline packages/functions-compat/package.json:1]: Looks like functions-compat also has a src/index.node.ts, and it's also referenced in this file's `test:node` npm script.

File: packages/functions-compat/package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "test:all": "run-p --npm-path npm test:browser test:node",
     "test:browser": "karma start --single-run",
     "test:browser:debug": "karma start --browsers=Chrome --auto-watch",
-    "test:node": "TS_NODE_COMPILER_OPTIONS='{\"module\":\"commonjs\"}' nyc --reporter lcovonly -- mocha 'src/{,!(browser)/**/}*.test.ts' --file src/index.node.ts --config ../../config/mocharc.node.js",
+    "test:node": "TS_NODE_COMPILER_OPTIONS='{\"module\":\"commonjs\"}' nyc --reporter lcovonly -- mocha 'src/{,!(browser)/**/}*.test.ts' --config ../../config/mocharc.node.js",
```
Comment (hsubox76) [inline packages/functions-compat/package.json:60]: Did you also remove the file itself? It's not showing up as removed in the Github diff for me.

File: packages/functions-compat/package.json [hunk 2]
```diff
@@ -3,16 +3,16 @@
...
   "module": "dist/index.esm2017.js",
   "esm5": "dist/index.esm5.js",
   "exports": {
     ".": {
       "types": "./dist/src/index.d.ts",
       "node": {
-        "import": "./dist/node-esm/index.node.esm.js",
-        "require": "./dist/index.node.cjs.js"
+        "require": "./dist/index.cjs.js",
+        "import": "./dist/index.esm2017.js"
```
Comment (DellaBitta) [inline packages/functions-compat/package.json:15]: I know this order is swapped, but now it matches the order of the browser's settings below. :)

File: packages/functions-compat/package.json (hunk 1 above)
Comment (DellaBitta) [reply packages/functions-compat/package.json:60]: Done!

File: packages/functions-compat/package.json
Comment (DellaBitta) [reply packages/functions-compat/package.json:1]: Removed.

**Review by hsubox76 — APPROVED** [review]

File: packages/functions/package.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
     "test:all": "run-p --npm-path npm test:browser test:node",
```
[the remark this answers is not in the document]
Comment (hsubox76) [reply packages/functions/package.json:38]: Oops, that's incorrect, we still need to run the tests in node. leave as is.

**Pull request merged**